test/ng/directive/ngTranscludeLocalsSpec.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+describe('ngTranscludeLocals', function() {
+
+  beforeEach(module(function($compileProvider) {
+      $compileProvider
+        .directive('transcludeParamDirective', function() {
+          return {
+            transclude: true,
+            scope: { param: '<' },
+            template: '<div ng-transclude ng-transclude-locals="param"></div>'
+          };
+        })
+        .component('transcludeListItemsComponent', {
+          transclude: true,
+          bindings: { param: '<' },
+          template: '<div ng-repeat="item in $ctrl.items track by $index">' +
+                      '<ng-transclude ng-transclude-locals="{item: item, position:$index+1, param:$ctrl.param}"></ng-transclude>' +
+                    '</div>',
+          controller: function() {
+            this.items = ['first','second','third'];
+          }
+        });
+    }));
+
+    var $compile, $rootScope;
+    beforeEach(inject(function(_$compile_,_$rootScope_) {
+      $compile = _$compile_;
+      $rootScope = _$rootScope_;
+    }));
+
+    var element;
+    afterEach(function() {
+      dealoc(element);
+    });
+
+    it('should set locals values', function() {
+      $rootScope.works = 'works';
+      element = $compile('<transclude-param-directive param="{it: works}">{{it}}</transclude-param-directive>')($rootScope);
+      $rootScope.$apply();
+
+      expect(element.text()).toBe('works');
+    });
+
+    it('should set locals values from objects', function() {
+      $rootScope.status = {is: 'success'};
+      element = $compile('<transclude-param-directive param="status">{{is}}</transclude-param-directive>')($rootScope);
+      $rootScope.$apply();
+
+      expect(element.text()).toBe('success');
+    });
+
+    it('should track changes in locals values', function() {
+      element = $compile('<transclude-param-directive param="status">{{partA}}-{{partB}}</transclude-param-directive>')($rootScope);
+
+      $rootScope.status = {partA: '1'};
+      $rootScope.$apply();
+      expect(element.text()).toBe('1-');
+
+      $rootScope.status.partB = '2';
+      $rootScope.$apply();
+      expect(element.text()).toBe('1-2');
+
+      delete $rootScope.status.partA;
+      $rootScope.$apply();
+      expect(element.text()).toBe('-2');
+
+      delete $rootScope.status;
+      $rootScope.$apply();
+      expect(element.text()).toBe('-');
+    });
+
+    it('should be compatible with ngRepeat', function() {
+      element = $compile('<transclude-list-items-component param="\'ctx\'">[{{position}}] {{item}} ({{param}})</transclude-list-items-component>')($rootScope);
+
+      $rootScope.$apply();
+      expect(element.text()).toBe('[1] first (ctx)[2] second (ctx)[3] third (ctx)');
+    });
+
+});
```

Comment:
I think we need more tests. E.g. what happens with multi-slot transclusion? What about optional, empty slots?